packages/salesforcedx-vscode-core/src/commands/forceStartApexDebugLogging.ts

```diff
@@ -0,0 +1,179 @@
+export let prevVFDebugLevel: string;
+export let debugLevelId: string;
+
+const MILLISECONDS_PER_SECOND = 60000;
+const LOG_TIMER_LENGTH_MINUTES = 30;
+
+export class ForceStartApexDebugLoggingExecutor extends SfdxCommandletExecutor<
+  TraceFlagInfo
+> {
+  public build(data: TraceFlagInfo): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_start_apex_debug_logging'))
+      .withArg('force:data:record:update')
+      .withFlag('--sobjecttype', 'DebugLevel')
+      .withFlag('--sobjectid', data.debugLevelId)
+      .withFlag('--values', 'ApexCode=Finest Visualforce=Finest')
+      .withArg('--usetoolingapi')
+      .build();
+  }
+
+  public execute(response: ContinueResponse<TraceFlagInfo>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        showTraceFlagExpiration(response.data.expirationDate);
+      }
+    });
+  }
+}
+
+export type TraceFlagInfo = {
+  traceflagId: string;
+  expirationDate: string;
+  debugLevelId: string;
+};
+class TraceFlagInfoGatherer implements ParametersGatherer<TraceFlagInfo> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<TraceFlagInfo>
+  > {
+    const execution = new CliCommandExecutor(
+      new SfdxCommandBuilder()
+        .withArg('force:data:soql:query')
+        .withFlag(
+          '--query',
+          'SELECT id, logtype, startdate, expirationdate, debuglevelid, debuglevel.apexcode, debuglevel.visualforce FROM TraceFlag'
+        )
+        .withArg('--usetoolingapi')
+        .withArg('--json')
+        .build(),
+      {
+        cwd: vscode.workspace.rootPath
+      }
+    ).execute();
+
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+    try {
+      const resultJson = JSON.parse(result);
+      if (resultJson.result.records.length > 0) {
+        const traceflag = resultJson.result.records[0];
+        developerLogDebugLevels.turnOnLogging(
+          traceflag.DebugLevelId,
+          traceflag.DebugLevel.ApexCode,
+          traceflag.DebugLevel.Visualforce
+        );
+        return {
+          type: 'CONTINUE',
+          data: {
+            traceflagId: traceflag.Id,
+            expirationDate: traceflag.ExpirationDate,
+            debugLevelId: traceflag.DebugLevelId
+          }
+        };
+      } else {
+        return { type: 'CANCEL' };
+      }
+    } catch (e) {
+      return { type: 'CANCEL' };
+    }
+  }
+}
+
+export class TraceFlagStartEndChecker
+  implements PostconditionChecker<TraceFlagInfo> {
```

Comment:
This a post condition checker since after getting the initial information, we should verify that the traceflag is actually active.